### DISCUSSION TITLE
monitoring camera status

### DISCUSSION THREAD

**Discussion by tve** [blakeblackshear/frigate#4050] [Ideas]
I'm having difficulties with some cameras on wifi in that they sometimes go out to lunch... i.e. stop providing a video feed. Sometimes it has to do with Wifi issues, sometimes, for my esp32-based cameras especially, it has to do with software lock-up. I'd like to set-up some alerting so at least I know that there's a problem and I can rectify it (or figure out how to do that automatically). The problem I'm having is that frigate doesn't provide a clear "camera is alive and feeding frames" signal...

Currently, there are two options:
- `/api/stats` has a `camera_fps` that presumably becomes zero if the camera is not sending frames. Problem 1: there's no information over which time-frame the fps are averaged. Problem 2: some of my esp32 cameras send frames very slowly and the rounding to one digit results in `0` even though the camera is alive.
- `/api/<camera>/latest.jpg` could be used to compare whether the frame is identical to the previous frame. Kind'a hokey.

Proposal:
- add a `frame_count` to `/api/stats` (or perhaps `camera_frames`, `process_frames`, `detection_frames`, `skipped_frames`) this is a typical way to enable monitoring of rates and can easily be fed into any monitoring system where alerts can then be set
- add a timestamp header to the `/api/<camera>/latest.jpg` response so code can tell how old that frame is, this would also be useful in the Web UI where I hate it when I look at the "cameras" overview page and everything looks fine, but in fact the image for one of the cameras is a couple of days old 'cause it broke: it would be nice to have a little yellow triangle in the UI with the age of the image...

Thoughts?

**Comment by thinkloop** [comment]
I second a need for clearer status alerting. Specifically for birdseye it would be nice if the black screen said something like "Monitoring 8 cameras, 2 inactive" or similar.

**Comment by evanheckert** [comment]
This is an indirect solution, but for general awareness of cameras or homelab services, I use Uptime Kuma. It's a simple open source tool that runs in docker, and will check for just about any type of response you could think of to verify that something is up, and notify you however you'd like if it's down. I do this with my cameras as well.

**Reply by thinkloop** [reply]
Thanks for the tip, looks very nice, gonna install it tonight 👍

**Reply by thinkloop** [reply]
May I ask what kind of monitor you setup in Uptime Kuma to monitor your cameras?

**Reply by evanheckert** [reply]
I can check later, but I believe it's just logging in as me on the camera's web UI. Not a great validation that it's streaming successfully, granted. I bet it had a way to test stream but haven't tried as of yet.

**Reply by LaurenceGough** [reply]
> I can check later, but I believe it's just logging in as me on the camera's web UI. Not a great validation that it's streaming successfully, granted. I bet it had a way to test stream but haven't tried as of yet.

I think my cameras would crash from logging in often into the web UI :D, it struggles enough as is haha!

**Comment by LaurenceGough** [comment]
I think it's a great idea, Frigate would feel so much more complete with a camera / feed down alert.

**Comment by tve** [comment]
Quick update:

> /api/stats has a camera_fps that presumably becomes zero if the camera is not sending frames.

This is of no use 'cause the camera_fps aren't updated if the camera doesn't send anything. So instead of dropping to zero it stays at the last value.

**Reply by NickM-27** [reply]
Use process_fps

**Reply by tve** [reply]
> Use process_fps

what makes you suggest that?

**Comment by jschwalbe** [comment]
I second this. Would be amazing to have something like Uptime Kuma be able to read (for example) `http://frigate.ip/api/stats/<camera_name>/available` (or via MQTT with a similar topic) and either get "online" or "offline".

**Comment by sebastian-tischer** [comment]
Hi team,
is there any nice solution available to check if a camera is still "live" / available? :)

**Reply by NickM-27** [reply]
yes, check the `camera_fps == 0` or if you are using HA integration 5.0 beta then you can just check if the camera entity is available or not

**Reply by sebastian-tischer** [reply]
Amazing buddy :) Does it make sense to already install 5.0 beta? Stable enough? :)

**Reply by NickM-27** [reply]
that version is meant to be installed alongside 0.13 beta of frigate

**Reply by sebastian-tischer** [reply]
I should have said I am using frigate in home assistant via HACS integration ;)

So should I install or wait until out of beta?

**Reply by NickM-27** [reply]
the integration is just the integration, you also have frigate installed somewhere. up to you, but I am not aware of any major or medium issues in either

**Reply by sebastian-tischer** [reply]
Damn, of course, installed as addon ;) would you recommend beta 2 as stable enough for production?

**Reply by NickM-27** [reply]
I personally find it stable enough

**Reply by sebastian-tischer** [reply]
Great thanks :) last question: can I install it besides version 12 or should I uninstall older version before?

**Reply by NickM-27** [reply]
it is simplest to only have one installed at a time

**Reply by sebastian-tischer** [reply]
Got it - I just installed version 0.13 and have two questions:

1.) If version 0.13. goes out of beta status, will it always be two separate addons or will you update the "normal" frigate addon (without the beta version in brackets)

2.) Sometimes I get an API error on the dashboard - I post it here, when I see it again :D

**Reply by sebastian-tischer** [reply]
{
  "request": {
    "type": "frigate/events/get",
    "instance_id": "frigate",
    "cameras": [
      "Eingang"
    ],
    "limit": 50,
    "id": 100
  },
  "response": {
    "code": "frigate_error",
    "message": "API error whilst retrieving events for cameras ['Eingang'] for Frigate instance frigate"
  }
}

**Reply by NickM-27** [reply]
1. All add-ons will be updated. 
2. Looks like the integration isn't updated

**Reply by sebastian-tischer** [reply]
I installed the 5.0 beta 2.0 and with that, I got the problems.

Besides that, my RPI keept freezing on a regular basis with both new addon and new integration. I just rolled back and all is finde again.

Are there any logs concerning the freeze I should provide you with?

**Reply by NickM-27** [reply]
No, there's really nothing we go off of with that, would need to know resource usage, system page screenshot, etc.
